File: imports-frontend-entities/src/entities/economic_operator_address.js
const handler = require('./base/handler')
const InternationalTelephone = require('./international_telephone')
const _ = require('lodash')

module.exports = class EconomicOperatorAddress {
  constructor(obj) {

    if (!obj) {
      obj = {}
    }

    this.addressLine1 = obj.addressLine1
    this.addressLine2 = obj.addressLine2
    this.addressLine3 = obj.addressLine3
    this.city = obj.city
    this.postalZipCode = obj.postalZipCode
    this.countryISOCode = obj.countryISOCode
    this.internationalTelephone = _.get(obj, 'internationalTelephone')
        ? new InternationalTelephone(obj.internationalTelephone) : undefined
    this.email = obj.email
    this.ukTelephone = obj.ukTelephone
    this.telephone = obj.telephone
    return Object.seal(new Proxy(this, handler))
  }
}

File: notification-schema-java/src/main/java/uk/gov/defra/tracesx/notificationschema/representation/EconomicOperatorAddress.java
package uk.gov.defra.tracesx.notificationschema.representation;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import uk.gov.defra.tracesx.notificationschema.validation.ErrorCodes;
import uk.gov.defra.tracesx.notificationschema.validation.annotations.RetrospectiveCloningProperty;
import uk.gov.defra.tracesx.notificationschema.validation.groups.NotificationCvedaFieldValidation;

@Builder
@Data
@JsonInclude(Include.NON_EMPTY)
@NoArgsConstructor(access = AccessLevel.PUBLIC)
@AllArgsConstructor(access = AccessLevel.PUBLIC)
public class EconomicOperatorAddress {

  @NotNull(groups = NotificationCvedaFieldValidation.class, message = ErrorCodes.NOT_NULL)
  @RetrospectiveCloningProperty()
  private String addressLine1;

  @RetrospectiveCloningProperty()
  private String addressLine2;

  @RetrospectiveCloningProperty()
  private String addressLine3;

  @NotNull(groups = NotificationCvedaFieldValidation.class, message = ErrorCodes.NOT_NULL)
  @RetrospectiveCloningProperty()
  private String city;

  @RetrospectiveCloningProperty()
  private String postalZipCode;

  @NotNull(groups = NotificationCvedaFieldValidation.class, message = ErrorCodes.NOT_NULL)
  @RetrospectiveCloningProperty()
  private String countryISOCode;

  private String ukTelephone;

  @Valid
  private InternationalTelephone internationalTelephone;

  @RetrospectiveCloningProperty()
  private String telephone;

  @RetrospectiveCloningProperty()
  private String email;
}
